pkg/transformers/image_test.go

```diff
@@ -251,7 +253,181 @@ func TestImageTransformer(t *testing.T) {
+			map[string]interface{}{
+				"spec": map[string]interface{}{
+					"template": map[string]interface{}{
+						"spec": map[string]interface{}{
+							"myContainers": []interface{}{
+								map[string]interface{}{
+									"name":  "nginx1",
+									"image": "nginx",
+								},
+							},
+						},
+					},
+				},
+				"spec2": map[string]interface{}{
+					"template": map[string]interface{}{
+						"spec": map[string]interface{}{
+							"myContainers": []interface{}{
+								map[string]interface{}{
+									"name":  "nginx3",
+									"image": "nginx:v1",
+								},
+								map[string]interface{}{
+									"name":  "nginx4",
+									"image": "my-nginx:latest",
+								},
+							},
+						},
+					},
+				},
+				"spec3": map[string]interface{}{
+					"template": map[string]interface{}{
+						"spec": map[string]interface{}{
+							"myInitContainers": []interface{}{
+								map[string]interface{}{
+									"name":  "postgresdb",
+									"image": "postgres:alpine-9",
+								},
+								map[string]interface{}{
+									"name":  "init-docker",
+									"image": "docker:17-git",
+								},
+								map[string]interface{}{
+									"name":  "myimage",
+									"image": "myprivaterepohostname:1234/my/image:latest",
+								},
+								map[string]interface{}{
+									"name":  "myimage2",
+									"image": "myprivaterepohostname:1234/my/image",
+								},
+								map[string]interface{}{
+									"name":  "my-app",
+									"image": "my-app-image:v1",
+								},
+								map[string]interface{}{
+									"name":  "my-cool-app",
+									"image": "gcr.io:8080/my-project/my-cool-app:latest",
+								},
+							},
+						},
+					},
+				},
+			}),
+	}
+	expected := resmap.ResMap{
+		resid.NewResId(gvk.Gvk{Kind: "Custom"}, "custom"): rf.FromMap(
+			map[string]interface{}{
+				"spec": map[string]interface{}{
+					"template": map[string]interface{}{
+						"spec": map[string]interface{}{
+							"myContainers": []interface{}{
+								map[string]interface{}{
+									"name":  "nginx1",
+									"image": "nginx:v2",
+								},
+							},
+						},
+					},
+				},
+				"spec2": map[string]interface{}{
+					"template": map[string]interface{}{
+						"spec": map[string]interface{}{
+							"myContainers": []interface{}{
+								map[string]interface{}{
+									"name":  "nginx3",
+									"image": "nginx:v2",
+								},
+								map[string]interface{}{
+									"name":  "nginx4",
+									"image": "my-nginx:previous",
+								},
+							},
+						},
+					},
+				},
+				"spec3": map[string]interface{}{
+					"template": map[string]interface{}{
+						"spec": map[string]interface{}{
+							"myInitContainers": []interface{}{
+								map[string]interface{}{
+									"name":  "postgresdb",
+									"image": "my-postgres:v3",
+								},
+								map[string]interface{}{
+									"name":  "init-docker",
+									"image": "my-docker@sha256:25a0d4b4a4c0eb97a1aabb8e29f18e917d05abfe1b7a7c07857230879ce7d3d3",
```

Comment:
ditto

Reply:
These are actually a copy and paste from existing code snippet. We may improve the readability in a separate PR.